actors/evm/src/interpreter/instructions/lifecycle.rs

```diff
@@ -1,17 +1,161 @@
-use crate::interpreter::address::EthAddress;
+use fil_actors_runtime::ActorError;
+use fil_actors_runtime::EAM_ACTOR_ADDR;
+use fvm_ipld_encoding::{strict_bytes, tuple::*, RawBytes};
+use fvm_shared::MethodNum;
+use fvm_shared::{address::Address, econ::TokenAmount};
+use serde_tuple::{Deserialize_tuple, Serialize_tuple};
+
+use crate::interpreter::stack::Stack;
+use crate::interpreter::{address::EthAddress, U256};
+use crate::state::State;
+
+use super::memory::{get_memory_region, MemoryRegion};
 use {
     crate::interpreter::{ExecutionState, StatusCode, System},
     fil_actors_runtime::runtime::Runtime,
     fvm_ipld_blockstore::Blockstore,
 };
 
+pub const CREATE_METHOD_NUM: u64 = 2;
+pub const CREATE2_METHOD_NUM: u64 = 3;
+
+#[derive(Serialize_tuple, Deserialize_tuple, Clone)]
+pub struct CreateParams {
+    #[serde(with = "strict_bytes")]
+    pub code: Vec<u8>,
+    pub nonce: u64,
+}
+
+#[derive(Serialize_tuple, Deserialize_tuple, Clone)]
+pub struct Create2Params {
+    #[serde(with = "strict_bytes")]
+    pub code: Vec<u8>,
+    pub salt: [u8; 32],
+}
+
+#[derive(Serialize_tuple, Deserialize_tuple, Debug, Clone, Copy, PartialEq, Eq)]
+pub struct EamReturn {
+    pub actor_id: u64,
+    pub robust_address: Address,
+    pub eth_address: EthAddress,
+}
+
 #[inline]
 pub fn create<'r, BS: Blockstore, RT: Runtime<BS>>(
-    _state: &mut ExecutionState,
-    _platform: &'r System<'r, BS, RT>,
-    _create2: bool,
+    state: &mut ExecutionState,
+    platform: &'r mut System<'r, BS, RT>,
 ) -> Result<(), StatusCode> {
-    todo!()
+    let ExecutionState { stack, memory, .. } = state;
+
+    // Overall future work / TODOs:
+    //  readonly state things | ISSUE NEEDED
```

Comment:
Likely an FVM thing. Can you file an issue? It shouldn't need any support from the EAM.